include/oneapi/dpl/pstl/omp/parallel_for.h

```diff
@@ -0,0 +1,55 @@
+namespace oneapi
+{
+namespace dpl
+{
+namespace __omp_backend
+{
+
+template <class _Index, class _Fp>
+void
+__parallel_for_body(_Index __first, _Index __last, _Fp __f)
+{
+    // initial partition of the iteration space into chunks
+    auto __policy = __omp_backend::__chunk_partitioner(__first, __last);
+
+    // To avoid over-subscription we use taskloop for the nested parallelism
+    _PSTL_PRAGMA(omp taskloop untied mergeable)
+    for (std::size_t __chunk = 0; __chunk < __policy.__n_chunks; ++__chunk)
+    {
+        __omp_backend::__process_chunk(__policy, __first, __chunk, __f);
+    }
+}
+
+//------------------------------------------------------------------------
+// Notation:
+// Evaluation of brick f[i,j) for each subrange [i,j) of [first, last)
+//------------------------------------------------------------------------
+
+template <class _ExecutionPolicy, class _Index, class _Fp>
+void
+__parallel_for(_ExecutionPolicy&&, _Index __first, _Index __last, _Fp __f)
+{
+    if (__should_run_serial(__first, __last))
```

Comment:
Okay. The Microsoft PSTL has gone the opposite way and only made things parallel when it looks like there's a benefit. I don't have a strong opinion on either approach, but it would be nice if we were consistent. See https://devblogs.microsoft.com/cppblog/using-c17-parallel-algorithms-for-better-performance/#current-limitations-of-the-msvc-implementation-of-parallel-algorithms